executor/chroot/chroot_executor.go

```diff
@@ -91,15 +89,29 @@ func (e *Executor) Execute(f def.Formula, j def.Job, d string) def.JobResult {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
-	if err := cmd.Start(); err != nil {
-		if err2, ok := err.(*exec.Error); ok && err2.Err == exec.ErrNotFound {
+	// launch execution.
+	// transform gosh's typed errors to repeatr's hierarchical errors.
+	var proc gosh.Proc
+	try.Do(func() {
+		proc = gosh.ExecProcCmd(cmd)
+	}).CatchAll(func(err error) {
+		switch err.(type) {
+		case gosh.NoSuchCommandError:
 			panic(executor.NoSuchCommandError.Wrap(err))
+		case gosh.NoArgumentsErr:
+			panic(executor.NoSuchCommandError.Wrap(err))
+		case gosh.ProcMonitorError:
+			panic(executor.TaskExecError.Wrap(err))
+		default:
+			panic(executor.UnknownError.Wrap(err))
 		}
-		panic(executor.TaskExecError.Wrap(err))
-	}
+		// FIXME all *this* also should actually be done in the barrier before returning a job, methinks?
+		// actually, not clear.  we haven't really spec'd that.  REVIEW.
+	}).Done()
 
 	// Wait for the job to complete
-	result.ExitCode = flak.WaitAndHandleExit(cmd)
+	// REVIEW: consider exposing `gosh.Proc`'s interface as part of repeatr's job tracking api?
+	result.ExitCode = proc.GetExitCode()
```

Comment:
Agreed.

My current plan is to land `use-mux-in-executors` after a rebase onto this PR, and then a new branch specifically to mess with Jobs. `use-mux-in-executors` will perpetuate or exacerbate the problem because I really want to mess with Jobs separately from mux upgrades, etc.